File: DesignPatternsExercises/ruby/CommandDesignPattern/db_extraction.rb
require_relative 'extraction'
require_relative 'db_extractor'

class DBExtraction < Extraction

  attr_accessor :file

  def initialize(file)
    @file = file
  end

  def call_correct_extraction
    # _________(7)______________
    DBExtractor.new.extract
  end

end

File: DesignPatternsExercises/java/CommandDesignPattern/Solution/DbExtraction.java
// class DbExtraction implements __________(7)__________ {
class DbExtraction implements Extraction {
	Extractor dbExtractor = new DbExtractor();

	@Override
	public void callCorrectExtraction() {
		// dbExtractor.__________(8)__________();
		dbExtractor.extract();
	}
}
